src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/MarshallingAttributeInfo.cs

```diff
@@ -536,62 +536,29 @@ private CountInfo CreateCountInfo(AttributeData marshalUsingData, ImmutableHashS
                 }
             }
 
-            if (!isArrayType)
+            if (isArrayType)
             {
-                return new MarshalAsInfo(unmanagedType, _defaultInfo.CharEncoding);
-            }
-
-            if (type is not IArrayTypeSymbol { ElementType: ITypeSymbol elementType })
-            {
-                _diagnostics.ReportConfigurationNotSupported(attrData, nameof(UnmanagedType), unmanagedType.ToString());
-                return NoMarshallingInfo.Instance;
-            }
-
-            MarshallingInfo elementMarshallingInfo = NoMarshallingInfo.Instance;
-            if (elementUnmanagedType != (UnmanagedType)SizeAndParamIndexInfo.UnspecifiedConstSize)
-            {
-                elementMarshallingInfo = new MarshalAsInfo(elementUnmanagedType, _defaultInfo.CharEncoding);
-            }
-            else
-            {
-                maxIndirectionDepthUsed = 1;
-                elementMarshallingInfo = GetMarshallingInfo(elementType, new Dictionary<int, AttributeData>(), 1, ImmutableHashSet<string>.Empty, ref maxIndirectionDepthUsed);
-            }
-
-            INamedTypeSymbol? arrayMarshaller;
+                if (type is not IArrayTypeSymbol { ElementType: ITypeSymbol elementType })
```

Comment:
Yet again this variable introduction is unhelpful and actively makes reviewing code outside of an editor much more difficult. Nothing to change just ranting.